### PR TITLE
#292 - Remove zero matrix check from LinearMap

### LINKED ISSUE #292: Remove zero matrix check from LinearMap
Outsourced from #290.
In `LinearMap.jl` we do this:
```julia
function *(M::AbstractMatrix, X::LazySet)
    @assert dim(X) == size(M, 2) "..."
    if findfirst(M) != 0
        return LinearMap(M, X)
    else
        return ZeroSet(dim(X))
    end
end
```
We should not always perform this check.

Idea: We could have some macro which goes one step further and sees that if the matrix is null obviously the outcome is a `ZeroSet`, e.g., `@simplify M * B` or even `@concrete` if we cannot do anything else with concrete `linear_map`.

### PR DESCRIPTION
**Pull request by mforets**
Closes #292.

### CODE REVIEW

File: src/LinearMap.jl [hunk 1]
```diff
@@ -22,12 +22,21 @@ type `N`, where `N` is a floating point number type such as `Float64`.
 struct LinearMap{NM, N} <: LazySet{N}
     M::AbstractMatrix{NM}
     X::LazySet{N}
+
+    # default constructor with dimension match check
+    function LinearMap{NM, N}(M::AbstractMatrix{NM}, X::LazySet{N}) where {NM, N}
+        @assert dim(X) == size(M, 2) "a linear map of size $(size(M)) cannot be applied to a set of dimension $(dim(X))"
+        return new(M, X)
```
Comment (schillic) [inline src/LinearMap.jl]: Do you not need `new{NM, N}(M, X)`?

File: src/LinearMap.jl [hunk 2]
```diff
@@ -44,14 +53,8 @@ Return the linear map of a convex set.
 
 If the matrix is null, a `ZeroSet` is returned; otherwise a lazy linear map.
```
Comment (schillic) [inline src/LinearMap.jl]: update docs

File: src/LinearMap.jl (hunk 1 above)
Comment (schillic) [reply src/LinearMap.jl]: Apparently not :hushed:

File: src/LinearMap.jl (hunk 2 above)
Comment (mforets) [reply src/LinearMap.jl]: oops, fixed.

**Pull request merged**